### PR TITLE
Init phetch

### PR DESCRIPTION
**Pull request by felixalbrigtsen**
###### Description of changes

Added package phetch(https://github.com/xvxx/phetch), a gopher client written in rust.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4365,6 +4365,13 @@
     githubId = 9959940;
     name = "Andreas Fehn";
   };
+  felixalbrigtsen = {
+    email = "felixalbrigtsen@gmail.com";
+    github = "felixalbrigtsen";
+    githubId = 64613093;
+    name = "Felix Albrigtsen";
+    matrix = "felixalb:pvv.ntnu.no";
```
Comment (dali99) [inline maintainers/maintainer-list.nix:4373]:
```suggestion
    matrix = "@felixalb:pvv.ntnu.no";
```

**Review by D4ndellion — CHANGES_REQUESTED** [review]

File: pkgs/applications/networking/gopher/phetch/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  nativeBuildInputs = [ pkg-config scdoc which ];
+  buildInputs = [ openssl ];
+
+  postInstall = ''
+    make manual
+    mkdir -p $out $man/share/man/man1
+    install -m644 doc/phetch.1 $man/share/man/man1
+  '';
+
+  doCheck = false;
```
Comment (dali99) [inline pkgs/applications/networking/gopher/phetch/default.nix:37]: The tests don't even compile, but the program works. You might want to add a comment as to why the tests are disabled.

**Review by D4ndellion — CHANGES_REQUESTED** [review]
You have to squash your commits together so that there are only two, one for adding the maintainer, and one for adding the package.

you can use interactive rebase and fixups to accomplish this

**Review by pbsds — APPROVED** [review]
LGTM

**Review by h7x4 — APPROVED** [review]

**Review by D4ndellion — APPROVED** [review]

File: pkgs/applications/networking/gopher/phetch/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  cargoSha256 = "sha256-y3Y5PnZ51Zc3LmVTijUGnb0KaGm28sWOSYxjuM3A1Zk=";
+
+  nativeBuildInputs = [ pkg-config scdoc which ];
+  buildInputs = [ openssl ] ++ lib.optionals stdenv.isDarwin [ Security ];
+
+  postInstall = ''
+    make manual
+    mkdir -p $out $man/share/man/man1
+    install -m644 doc/phetch.1 $man/share/man/man1
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/gopher/phetch/default.nix:33]:
```suggestion
    installMagPage doc/phetch.1
```

File: pkgs/applications/networking/gopher/phetch/default.nix [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      - No-nonsense keyboard navigation
+      - Supports Gopher searches, text and menu pages, and downloads
+      - Save your favorite Gopher sites with bookmarks
+      - Opt-in history tracking
+      - Secure Gopher support (TLS)
+      - Tor support
+    '';
+    changelog = "https://github.com/xvxx/phetch/releases/tag/v${version}";
+    homepage = "https://github.com/xvxx/phetch";
+    license = "licenses.mit";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/gopher/phetch/default.nix:53]:
```suggestion
    license = licenses.mit;
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**